src/uucore/signals.rs

```diff
@@ -325,21 +325,69 @@ pub static ALL_SIGNALS: [Signal<'static>; 31] = [
 ];
 
 pub fn signal_by_name_or_value(signal_name_or_value: &str) -> Option<usize> {
-    if signal_name_or_value == "0" {
-        return Some(0);
-    }
-    for signal in &ALL_SIGNALS {
-        let long_name = format!("SIG{}", signal.name);
-        if signal.name == signal_name_or_value || (signal_name_or_value == signal.value.to_string())
-            || (long_name == signal_name_or_value)
-        {
-            return Some(signal.value);
+    if let Ok(value) = signal_name_or_value.parse() {
+        if is_signal(value) {
+            return Some(value);
+        } else {
+            return None;
         }
     }
-    None
+    let signal_name = if signal_name_or_value.starts_with("SIG") {
```

Comment:
I think this can just be `let signal_name = signal_name_or_values.trim_left_matches("SIG");`